hibike/hibike_communicator.py

```diff
@@ -0,0 +1,218 @@
+"""
+   This library contains a set of functions that allow the user to manually send and recieve  messages to and from a device by calling functions
+"""
+
+import hibike_message as hm
+import hibike_process as hp
+import multiprocessing
+import threading
+import time
+
+class HibikeCommunicator:
+  
+   def __init__(self):
+      """
+          Set up the pipes for communication between the device and the Beegle Bum Black, creates a thread to recieve communications from the device, starts up the process that runs the communication, and sets up a dictionary that caches all device values sent to it.
+      """
+      
+      # This block creates the pipes
+      self.pipeToChild, pipeFromChild = multiprocessing.Pipe()
+      
+      # This block creates the process
+      badThingsQueue = multiprocessing.Queue()
+      self.stateQueue = multiprocessing.Queue()
+      newProcess = multiprocessing.Process(target=hp.hibike_process, name="hibike_sim", args=[badThingsQueue,self.stateQueue, pipeFromChild])
+      newProcess.daemon = True
+      newProcess.start()
+      self.pipeToChild.send(["enumerate_all", []])
+      
+      # Creates the list of uids
+      self.uids = set()
+      
+      # This block creates the thread
+      outThread =  threading.Thread(target = self.process_output)
+      outThread.start()
+   
+      # Create the device values dictionary cache
+      self.device_values_cache = {}
+
+   def process_output(self):
+      """
+         Prints out messages from the devices that are uploaded by newProcess to stateQueue
+         If it's a subscription response from a device whose uid is not in self.uids, the uid will be added to self.uids
+         If it's a device disconnection from a device whose uid in self.uids, the uid will be removed from self.uids
+      """
+      while True:
+         output = self.stateQueue.get()
+         print(output)
```

Comment:
In that case, should I create another function that prints it, or can I just leave this printing functionality out of the library?